src/output/details.rs

```diff
@@ -228,23 +231,19 @@ impl<'a> Render<'a> {
                     // that they want to see them.
 
                     if xattr::ENABLED {
-                        match file.path.attributes() {
-                            Ok(xs) => {
-                                xattrs.extend(xs);
-                            }
-                            Err(e) => {
-                                if self.opts.xattr {
-                                    errors.push((e, None));
-                                }
-                                else {
-                                    error!("Error looking up xattr for {:?}: {:#?}", file.path, e);
-                                }
-                            }
-                        }
+                        xattrs.extend(&file.extended_attributes);
                     }
 
+                    // Do not show the hint '@' if the only extended attribute is the security
+                    // attribute and the security attribute column is active.
+                    let show_xattr_hint = xattrs.len() > 1 || (xattrs.len() == 1 && (
+                        match file.security_context().context {
+                            SecurityContextType::SELinux(_) => false,
+                            SecurityContextType::None       => true }
+                         || !self.opts.secattr));
```

Comment:
I believe something like that is much clearer:

```suggestion
                    let has_selinux_ctx = match file.security_context().context {
                        SecurityContextType::SELinux(_) => true,
                        _ => false,
                    };
                    let show_xattr_hint = xattrs.len() > 1 || (xattrs.len() == 1 && (!has_selinux_ctx || !self.opts.secattr));
```